apps/dailyscrum/src/app/orgs/[orgId]/add-scrum-update-dialog.tsx

```diff
@@ -0,0 +1,366 @@
+
+type DynamicFormValues = { [x: string]: string };
+type FormValues = { date: Date };
+
+const DateField: React.FC<{
+  field: ControllerRenderProps<
+    {
+      date: Date;
+    },
+    "date"
+  >;
+  today: DateTime<true> | DateTime<false>;
+  timeZone: string;
+}> = ({ field, today, timeZone }) => {
+  const selected = useMemo(() => field.value, [field.value]);
+
+  const tomorrow = useMemo(() => today.plus({ days: 1 }), [today]);
+
+  const formattedDate = useMemo(() => {
+    return DateTime.fromJSDate(selected)
+      .setLocale("en-US")
+      .toLocaleString(DateTime.DATE_FULL);
+  }, [selected]);
+
+  const handleSelect = useCallback(
+    (systemZoneDate: Date | undefined) => {
+      if (!systemZoneDate) {
+        return;
+      }
+
+      const date = DateTime.fromJSDate(systemZoneDate).setZone(timeZone, {
+        keepLocalTime: true,
+      });
+
+      field.onChange(date.toJSDate());
+    },
+    [field, timeZone]
+  );
+
+  const disabled = useCallback(
+    (systemZoneDate: Date) => {
+      const date = DateTime.fromJSDate(systemZoneDate).setZone(timeZone, {
+        keepLocalTime: true,
+      });
+
+      return date < today || date > tomorrow;
+    },
+    [today, tomorrow, timeZone]
+  );
+
+  return (
+    <FormItem className="flex flex-col">
+      <FormLabel>Date</FormLabel>
+      <FormControl>
+        <Popover>
+          <PopoverTrigger asChild>
+            <FormControl>
+              <Button
+                variant={"outline"}
+                size="sm"
+                className="w-[240px] font-normal"
+              >
+                {formattedDate}
+                <CalendarIcon className="ml-auto h-4 w-4 opacity-50" />
+              </Button>
+            </FormControl>
+          </PopoverTrigger>
+          <PopoverContent className="w-auto p-0" align="start">
+            <Calendar
+              mode="single"
+              selected={selected}
+              onSelect={handleSelect}
+              disabled={disabled}
+              initialFocus
+            />
+          </PopoverContent>
+        </Popover>
+      </FormControl>
+      <FormDescription>
+        Select a date for your daily scrum update. You can choose either today
+        or tomorrow.
+      </FormDescription>
+      <FormMessage />
+    </FormItem>
+  );
+};
+
+type Question = {
+  id: string;
+  label: string;
+  placeholder?: string;
+  description?: string;
+  isRequired: boolean;
+  maxLength?: number;
+};
+
+export interface AddScrumUpdateDialogProps {
+  dailyScrumUpdateFormId: number;
+  description: string;
+  questions: Question[];
+  timeZone: string;
+  hasAddedDailyScrumUpdateForToday: boolean;
+  hasAddedDailyScrumUpdateForTomorrow: boolean;
+}
+
+export const AddScrumUpdateDialog: React.FC<AddScrumUpdateDialogProps> = ({
+  dailyScrumUpdateFormId,
+  description,
+  questions,
+  timeZone,
+  hasAddedDailyScrumUpdateForToday,
+  hasAddedDailyScrumUpdateForTomorrow,
+}) => {
```

Comment:
Added dialog component

Reply:
https://github.com/kiwiy-is/dailyscrum/pull/32/files/405a65ea9a37c7a414947657aecab94da5247ec7#r1519331097

Reply:
https://github.com/kiwiy-is/dailyscrum/pull/32/files/405a65ea9a37c7a414947657aecab94da5247ec7#r1519284621

Reply:
https://github.com/kiwiy-is/dailyscrum/pull/32/files#r1519280152